src/service/cia301/co_csdo.c

```diff
@@ -428,6 +465,214 @@ static CO_ERR COCSdoFinishDownloadSegmented(CO_CSDO *csdo)
     COCSdoTransferFinalize(csdo);
     return CO_ERR_SDO_SILENT;
 }
+/******************************************************************************
+* Block Upload Transfer Functions 
+* ****************************************************************************/
+// cbt TODO: Implement this function
+static CO_ERR COCSdoInitUploadBlock        (CO_CSDO *csdo);
+// cbt TODO: Implement this function
+static CO_ERR COCSdoUploadSubBlock         (CO_CSDO *csdo);
+
+/******************************************************************************
+* Block Download Transfer Functions 
+* ****************************************************************************/
+// Set up client side after getting confirmation from the server
+static CO_ERR COCSdoInitDownloadBlock      (CO_CSDO *csdo)
+{
+    CO_ERR    result = CO_ERR_SDO_SILENT;
+    uint16_t  Idx;
+    uint8_t   Sub;
+    uint8_t cmd;
+    CO_IF_FRM frm;
+
+    cmd = CO_GET_BYTE(csdo->Frm, 0u);
+
+    Idx = CO_GET_WORD(csdo->Frm, BLOCK_DOWNLOAD_FRM_INIT_RESPONSE_IDX_BYTE_OFFSET);
+    Sub = CO_GET_BYTE(csdo->Frm, BLOCK_DOWNLOAD_FRM_INIT_RESPONSE_SUB_BYTE_OFFSET);
+
+    // makes no difference
+     /* clean frm data */
+    CO_SET_LONG(&frm, 0, 0u);
+    CO_SET_LONG(&frm, 0, 4u);
+
+    // verfiy block transfer from the correct Idx and Sub
+    uint8_t cmd_scs = READ_BITS(cmd, BLOCK_DOWNLOAD_CMD_SCS_CCS_BIT_OFFSET, BLOCK_DOWNLOAD_CMD_SCS_CCS_BIT_MASK);
+    if ((cmd_scs == BLOCK_DOWNLOAD_CMD_SCS) && 
+        (Idx == csdo->Tfer.Idx) &&
+        (Sub == csdo->Tfer.Sub)) {
+
+        // uint8_t cmd_cc = READ_BITS(cmd, BLOCK_DOWNLOAD_INIT_REQUEST_CMD_CRC_BIT_OFFSET, 0x01);
```

Comment:
Is this staying? I assume this has something to do with the CRC. If so, just tag it with the CRC Jira ticket